### PR TITLE
feat(cmd/network): Autodetect chain context if none passed

### PR DESCRIPTION
**Pull request by amela**

### CODE REVIEW

File: cmd/network/set_chain_context.go [hunk 1]
```diff
@@ -1,30 +1,59 @@
...
 
 	cliConfig "github.com/oasisprotocol/cli/config"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/connection"
+)
+
+var (
+	chainContext string
+
+	setChainContextCmd = &cobra.Command{
+		Use:   "set-chain-context <name>",
```
Comment (matevz) [inline cmd/network/set_chain_context.go:20]: The chain context is optional, but it's still there. Check out how we do it in the [`oasis account withdraw`](https://github.com/oasisprotocol/cli/blob/master/cmd/account/withdraw.go#L22) command.

```suggestion
		Use:   "set-chain-context <name> [chain-context]",
```

File: cmd/network/set_chain_context.go [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
-
-		net.ChainContext = chainContext
-
-		err := cfg.Save()
-		cobra.CheckErr(err)
-	},
+func init() {
+	chainContextFlag := flag.NewFlagSet("", flag.ContinueOnError)
+	chainContextFlag.StringVar(&chainContext, "chain-context", "", "set chain-context")
+	setChainContextCmd.Flags().AddFlagSet(chainContextFlag)
```
Comment (matevz) [inline cmd/network/set_chain_context.go:58]: Nah, no need for a separate flag. Just put the chain-context as an (optional) positional argument.

File: cmd/network/set_chain_context.go [hunk 3]
```diff
@@ -1,28 +1,46 @@
 package network
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/spf13/cobra"
 
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
```
Comment (kostko) [inline cmd/network/set_chain_context.go:9]: Please group these correctly.

**Review by kostko — COMMENTED** [review]
Mostly looks good, one minor nit.

File: docs/network.md [hunk 4]
```diff
@@ -110,12 +110,18 @@ documentation.
...
-![code shell](../examples/network/05-set-chain-context.in)
+![code shell](../examples/network/05-set-chain-context-ctx.in)
 
 ![code shell](../examples/network/06-list.in)
 
 ![code shell](../examples/network/06-list.out)
 
+To automatically detect the chain context, simply run the command without the
+`[chain-context]` argument. This is especially useful for Localnet, where the
+chain context often changes each time you restart the `oasis-net-runner`.
```
Comment (matevz) [inline docs/network.md:121]:
```suggestion
chain context changes each time you restart the `oasis-net-runner`:
```

File: docs/network.md [hunk 5]
```diff
@@ -110,12 +110,18 @@ documentation.
...
 ![code shell](../examples/network/04-list.out)
 
-![code shell](../examples/network/05-set-chain-context.in)
+![code shell](../examples/network/05-set-chain-context-ctx.in)
 
 ![code shell](../examples/network/06-list.in)
 
 ![code shell](../examples/network/06-list.out)
 
+To automatically detect the chain context, simply run the command without the
```
Comment (matevz) [inline docs/network.md:119]:
```suggestion
To automatically detect the chain context, simply omit the
```

**Review by matevz — APPROVED** [review]
Works, thanks.

**Pull request merged**